tracing/options.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2022 CloudWeGo Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracing
+
+import (
+	"context"
+
+	"github.com/cloudwego/kitex/pkg/remote"
+	"github.com/cloudwego/kitex/pkg/rpcinfo"
+	"github.com/cloudwego/kitex/pkg/transmeta"
+	"github.com/cloudwego/kitex/transport"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/global"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	instrumentationName = "github.com/kitex-contrib/obs-opentelemetry"
+)
+
+// Option opts for opentelemetry tracer provider
+type Option interface {
+	apply(cfg *config)
+}
+
+type option func(cfg *config)
+
+func (fn option) apply(cfg *config) {
+	fn(cfg)
+}
+
+type config struct {
+	tracer trace.Tracer
+	meter  metric.Meter
+
+	tracerProvider    trace.TracerProvider
+	meterProvider     metric.MeterProvider
+	textMapPropagator propagation.TextMapPropagator
+
+	spanNameFormatter func(ctx context.Context) string
+
+	withStackTrace        bool
+	recordSourceOperation bool
+
+	kitexTransportProtocol transport.Protocol
+	kitexClientMetaHandler remote.MetaHandler
+	kitexServerMetaHandler remote.MetaHandler
```

Comment:
These field is unused

Reply:
Yes, I forgot to remove it, I'll change it